reference/7/Microsoft.PowerShell.Core/About/about_Pipeline_Chain_Operators.md

```diff
@@ -0,0 +1,291 @@
+with cmdlets or functions. For example:
+
+```powershell
+# Create an SSH key pair - if successful copy the public key to clipboard
+ssh-keygen -t rsa -b 2048 && Get-Content -Raw ~\.ssh\id_rsa.pub | clip
+```
+
+### Examples
+
+#### Two successful commands
+
+```powershell
+Write-Output 'First' && Write-Output 'Second'
+```
+
+```Output
+First
+Second
+```
+
+#### First command fails, causing second not to be executed
+
+```powershell
+Write-Error 'Bad' && Write-Output 'Second'
+```
+
+```Output
+Write-Error 'Bad' && Write-Output 'Second' : Bad
++ CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
++ FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException
+```
+
+#### First command succeeds, so the second command is not executed
+
+```powershell
+Write-Output 'First' || Write-Output 'Second'
+```
+
+```Output
+First
+```
+
+#### First command fails, so the second command is executed
+
+```powershell
+Write-Error 'Bad' || Write-Output 'Second'
+```
+
+```Output
+Write-Error 'Bad' && Write-Output 'Second' : Bad
++ CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
++ FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException
+
+Second
+```
+
+Pipeline success is defined by the value of the `$?` variable, which PowerShell
+automatically sets after executing a pipeline based on its execution status.
+This means that pipeline chain operators have the following equivalence:
+
+```powershell
+Test-Command '1' && Test-Command '2'
+```
+
+works the same as
+
+```powershell
+Test-Command '1'; if ($?) { Test-Command '2' }
+```
+
+and
+
+```powershell
+Test-Command '1' || Test-Command '2'
+```
+
+works the same as
+
+```powershell
+Test-Command '1'; if (-not $?) { Test-Command '2' }
+```
+
+### Assignment from pipeline chains
+
+Assigning a variable from a pipeline chain takes the concatenation of all the
+pipelines in the chain:
+
+```powershell
+$result = Write-Output '1' && Write-Output '2'
+$result
+```
+
+```Output
+1
+2
+```
+
+If a script-terminating error is thrown during assignment from a pipeline
+chain, the assignment does not succeed:
+
+```powershell
+try
+{
+    $result = Write-Output 'Value' && $(throw 'Bad')
+}
+catch
+{
+    # Do nothing, just squash the error
+}
+
+"Result: $result"
+```
+
+```Output
+Result:
+```
+
+### Operator syntax and precedence
+
+Unlike other operators, `&&` and `||` operate on pipelines, rather than on
+expressions like `+` or `-and`, for example.
+
+`&&` and `||` have a lower precedence than piping (`|`) or redirection (`>`),
+but a higher precedence than job operators (`&`), assignment (`=`) or
+semicolons (`;`). This means that pipelines within a pipeline chain can be
+individually redirected, and that entire pipeline chains can be backgrounded,
+assigned to variables, or separated as statements.
+
+To use lower precedence syntax within a pipeline chain, consider the use of
+parentheses `(...)` or a subexpression `$(...)`.
```

Comment:
I omitted a thing here about `&&` and `||` being left associative (so `cmd1 && cmd2 && cmd3` is the same as `[cmd1 && cmd2] && cmd3`), but wasn't sure if that's too technical here.

It's relevant if you do something like:

```powershell
Get-Item ./file.txt || Write-Error 'File not found' && Get-Content -Raw ./file.txt
```

The associativity means that `Get-Content` will be executed based on the result of `Get-Item || Write-Error`, so if `Get-Item` succeeds it is executed, but if it doesn't `Write-Error` runs, which by definition does not succeed, so `Get-Content` does not succeed.

Reply:
I think it would be good to a note about associativity. Can associativity be altered using paretheses or braces? For example:

```
(Get-Item ./file.txt || Write-Error 'File not found') && Get-Content -Raw ./file.txt
{Get-Item ./file.txt || Write-Error 'File not found'} && Get-Content -Raw ./file.txt
```

Reply:
Parentheses currently don't work nicely because they always set `$?` to `$true`... That's always been the case unfortunately, although this feature has many people talking about changing it.

Unfortunately the dot-sourced scriptblock also suffers from this.

Reply:
Ok, I've added a note about associativity, plus a bit about `(...)` etc setting `$?`

Reply:
Looks good. Thanks